pkg/reconciler/knservice/resources/knative_service.go

```diff
@@ -0,0 +1,54 @@
+package resources
+
+import (
+	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/containers/tensorflow"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func getModelServingContainer(modelSpec *v1alpha1.ModelSpec, modelName string) *v1.Container {
+	if modelSpec.Tensorflow != nil {
+		return tensorflow.CreateTensorflowContainer(modelSpec.Tensorflow, modelName)
+	} else {
+		//TODO(@yuzisun) handle other model types
+		return &v1.Container{}
+	}
+}
+
+func CreateKnService(desiredService *v1alpha1.KFService) (*knservingv1alpha1.Service, error) {
+	var revisions []string
+	container := &v1.Container{}
+	routingPercent := int32(0)
+	if desiredService.Spec.Canary == nil ||
+		(desiredService.Spec.Canary.TrafficPercent == 0 && desiredService.Spec.Canary != nil) {
```

Comment:
You don't need the check desiredService.Spec.Canary != nil check.

Reply:
Does knative not allow for canarying 0% traffic? I was hoping this might be a little more transparent to knative

Reply:
I guess you are right, 0 is allowed. I think I can remove this check let me test out.